proposals/span-safety.md

```diff
@@ -0,0 +1,102 @@
+# Compile time enforcement of safety for ref-like types #
+
+The main reason for the additional safety rules when dealing with types like `Span<T>` is that such types must be confined to the execution stack.
+ 
+There are three reasons why `Span<T>` must be a stack-only type.
+
+1. `Span<T>` is basically a struct containing `{reference to data, offset limit}`. Concurrent "tearing" of such struct would lead to the possibility of out of range accesses and type-safety violations, which basically means "GC heap corruption".
```

Comment:
I would give the tuple example here.  Essentially: `Span<T>` is logically a `(ref T data, int length)` structure.